Mathlib/FieldTheory/JacobsonNoether.lean

```diff
@@ -0,0 +1,198 @@
+  greater than 0 such that `(a * x - x * a) ^ n = 0` (as linear maps) for
+  every `n` greater than `(p ^ m)`. -/
+lemma exist_pow_eq_zero_of_le (p : ℕ) [hchar : ExpChar D p]
+    {a : D} (ha : a ∉ k) (hinsep : ∀ x : D, IsSeparable k x → x ∈ k):
+  ∃ m ≥ 1, ∀ n ≥ (p ^ m), (δ a) ^ n = 0 := by
+  obtain ⟨m, hm⟩ := exists_pow_mem_center_of_inseparable' p ha hinsep
+  refine ⟨m, ⟨hm.1, fun n hn ↦ ?_⟩⟩
+  have inter : (δ a) ^ (p ^ m) = 0 := by
+    ext x
+    rw [δ, Pi.sub_apply, sub_pow_expChar_pow_of_commute p m (commute_mulLeft_right a a)]
+    rw [sub_apply, l_pow, r_pow, sub_eq_zero_of_eq]; rfl
+    suffices h : a ^ (p ^ m) ∈ k from (Subring.mem_center_iff.1 h x).symm
+    exact hm.2
+  rw [((Nat.sub_eq_iff_eq_add hn).1 rfl), pow_add, inter, mul_zero]
+
+variable (D) in
+/-- For a non-commutative finite dimensional division algebra D (with base ring being its center),
+  there exist an element `x` that is separable over its center-/
+theorem Jacobson_Noether (H : k ≠ (⊤ : Subring D)) :
+    ∃ x : D, x ∉ k ∧ IsSeparable k x := by
+  obtain ⟨p, hp⟩ := ExpChar.exists D
+  by_contra! insep
+  have hinsep : ∀ x : D, IsSeparable k x → x ∈ k :=
+    fun x h ↦ Classical.byContradiction fun hx ↦ insep x hx h
+  -- The element `a` below is in `D` but not in `k`.
+  obtain ⟨a, ha⟩ := not_forall.mp <| mt (Subring.eq_top_iff' k).mpr H
+  have ha₀ : a ≠ 0 := fun nh ↦ nh ▸ ha <| Subring.zero_mem k
+  -- We construct another element `b` that does not commute with `a`.
+  obtain ⟨b, hb1⟩ : ∃ b : D , (δ a) b ≠ 0 := by
+    rw [Subring.mem_center_iff, not_forall] at ha
+    use ha.choose
+    show a * ha.choose - ha.choose * a ≠ 0
+    simpa only [ne_eq, sub_eq_zero] using Ne.symm ha.choose_spec
+  -- We find a maximum natural number `n` such that `(δ a) ^ n b ≠ 0`.
+  obtain ⟨n, hn, hb⟩ : ∃ n > 0, ((δ a) ^ n) b ≠ 0 ∧ ((δ a) ^ (n + 1)) b = 0 := by
+    obtain ⟨m, -, hm2⟩ := exist_pow_eq_zero_of_le p ha hinsep
+    have exist : ∃ n > 0, ((δ a) ^ (n + 1)) b = 0 := by
+      refine ⟨(p ^ m), ⟨expChar_pow_pos D p m, by rw [hm2 (p ^ m + 1) (by omega)]; rfl⟩⟩
+    classical
+    refine ⟨Nat.find exist, ⟨(Nat.find_spec exist).1, ?_, (Nat.find_spec exist).2⟩⟩
+    set t := (Nat.find exist - 1 : ℕ) with ht
+    by_cases choice : 0 < t
+    · have := @Nat.find_min (H := exist) _ _ t ?_
+      · exact (@Nat.sub_add_cancel (Nat.find exist) 1 (by omega) ▸ ht ▸ not_and.1 this) choice
+      · exact Nat.sub_one_lt <| ne_of_gt (Nat.find_spec exist).1
+    · rw [not_lt, Nat.le_zero] at choice
+      have := Nat.eq_add_of_sub_eq (Nat.find_spec exist).1 ht.symm
+      simp only [gt_iff_lt, choice, Nat.succ_eq_add_one, zero_add] at this
+      rw [this, pow_one]
+      exact hb1
+    -- We define `c` to be the value that we proved above to be non-zero.
+  set c := ((δ a) ^ n) b with hc_def
+  letI : Invertible c := ⟨c⁻¹, inv_mul_cancel₀ (hb.1), mul_inv_cancel₀ (hb.1)⟩
+  -- We prove that `c` is commute with `a`.
+  have hc : c * a = a * c := by
+    symm; apply eq_of_sub_eq_zero
```

Comment:
This `symm` shows that you set up your equations in the wrong order (both `hc` and `hb.2`): better change them and avoid this `symm` here.

Reply:
(Sorry for the delay, was dealing with some school stuff these two days)
Indeed, I've reverted ```hc```. Regarding ```hb.2```, should we change that too? Given that both left and right parts of ```hb``` are used.